replit_river/codegen/client.py

```diff
@@ -209,8 +207,8 @@ def flatten_union(tpe: RiverType) -> list[RiverType]:
                         typeddict_encoder.append(f"encode_{type_name}(x)")
                     typeddict_encoder.append(
                         f"""
-                            if x['{discriminator_name}']
-                            == '{discriminator_value}'
+                            if x[{repr(discriminator_name)}]
+                            == {repr(discriminator_value)}
```

Comment:
huh, how did this even work lol

Reply:
It's equivalent in the happy path but `repr` will surface bugs in the not happy path